planning/behavior_path_planner/src/planner_manager.cpp

```diff
@@ -218,14 +202,47 @@ std::vector<SceneModulePtr> PlannerManager::getRequestModules(
     stop_watch_.tic(manager_ptr->name());
 
     /**
-     * don't launch candidate module if approved modules already exist.
+     * determine the execution capability of modules based on existing approved modules.
      */
-    if (!approved_module_ptrs_.empty()) {
-      if (!manager_ptr->isSimultaneousExecutableAsApprovedModule()) {
+    const bool has_non_always_executable_module = std::any_of(
+      approved_module_ptrs_.begin(), approved_module_ptrs_.end(),
+      [this](const auto & m) { return !getManager(m)->isAlwaysExecutableModule(); });
+    if (has_non_always_executable_module) {
+      // Condition 1: do not add modules that are neither always nor simultaneous executable
+      // if there exists at least one approved module that is simultaneous but not always
+      // executable. (only modules that are either always executable or simultaneous executable can
+      // be added)
+      const auto find_non_always_simultaneous_module = [this](const auto & m) {
```

Comment:
```suggestion
      const auto find_block_module = [this](const auto & m) {
```

[imo] `find_non_always_simultaneous_module` is not understandable. I think it is better to use `find_block_module`.

Reply:
ok!
used `find_block_module` also in Condition 2, so add unnamed namespace?

```
    // Condition 2: Only modules that are always executable can be added
    // if there exists at least one executable module that is neither always nor simultaneous
    // executable.
    const auto find_block_module = [this](const auto & m) {
      return !getManager(m)->isAlwaysExecutableModule() &&
             !getManager(m)->isSimultaneousExecutableAsCandidateModule();
    };
```